tests/unit/amo/components/TestConfirmButton.js

```diff
@@ -27,138 +22,147 @@ describe(__filename, () => {
+    expect(dialogCancelButton).toHaveClass('Button--puffy');
+
+    const dialogConfirmButton = screen.getByRole('button', {
+      name: confirmButtonText,
+    });
+    expect(dialogConfirmButton).toHaveClass(
+      'ConfirmationDialog-confirm-button',
+    );
+    expect(dialogConfirmButton).toHaveClass('Button--alert');
+    expect(dialogConfirmButton).toHaveClass('Button--puffy');
   });
 
   it('hides the default button after it is clicked', () => {
-    const { store } = dispatchClientMetadata();
-    const root = render({ store });
-
-    root
-      .find('.ConfirmButton-default-button')
-      .simulate('click', createFakeEvent());
-    applyUIStateChanges({ root, store });
+    render();
 
-    expect(root.find('.ConfirmButton-default-button')).toHaveLength(0);
+    const button = screen.getByRole('button', {
+      name: 'the default text of this button',
+    });
+    userEvent.click(button);
+    expect(button).not.toBeInTheDocument();
   });
 
-  it('hides ConfirmationDialog via onCancel', () => {
-    const { store } = dispatchClientMetadata();
-    const root = renderWithDialog({ store });
+  it('handles onCancel callback and hides ConfirmationDialog on cancel', () => {
+    const onCancel = jest.fn();
+    renderWithDialog({ onCancel });
+
+    expect(
+      screen.getByClassName('ConfirmButton--show-confirmation'),
+    ).toBeInTheDocument();
 
-    const onCancel = getDialogProp(root, 'onCancel');
-    // Simulate a cancellation.
-    onCancel(createFakeEvent());
+    userEvent.click(screen.getByClassName('ConfirmationDialog-cancel-button'));
 
-    applyUIStateChanges({ root, store });
-    expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
-    expect(root.find(ConfirmationDialog)).toHaveLength(0);
+    expect(
+      screen.queryByClassName('ConfirmButton--show-confirmation'),
+    ).not.toBeInTheDocument();
+    expect(onCancel).toHaveBeenCalled();
   });
 
-  it('handles the onConfirm callback', () => {
-    const { store } = dispatchClientMetadata();
-    const onConfirmSpy = sinon.spy();
-    const root = renderWithDialog({ onConfirm: onConfirmSpy, store });
+  it('handles onConfirm callback and hides ConfirmationDialog on confirm', () => {
+    const onConfirm = jest.fn();
+    renderWithDialog({ onConfirm });
 
-    const onConfirm = getDialogProp(root, 'onConfirm');
+    expect(
+      screen.getByClassName('ConfirmButton--show-confirmation'),
+    ).toBeInTheDocument();
 
-    // Simulate a confirmation.
-    const event = createFakeEvent();
-    onConfirm(event);
-    applyUIStateChanges({ root, store });
+    userEvent.click(screen.getByClassName('ConfirmationDialog-confirm-button'));
 
-    expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
-    expect(root.find(ConfirmationDialog)).toHaveLength(0);
-    sinon.assert.calledWith(onConfirmSpy, event);
+    expect(
+      screen.queryByClassName('ConfirmButton--show-confirmation'),
```

Comment:
See above. There should be another way to assert we are no longer on the confirmation dialog screen.